### PR TITLE
PA-5571 Improve help messaging / override default

### PR DESCRIPTION
**Pull request by SOOS-JAlvarez**
### Changes:
  - Improve help messaging - display it in table format

**Ticket:** https://soos.atlassian.net/browse/PA-5571

### CODE REVIEW

File: main.py [hunk 1]
```diff
@@ -881,6 +891,21 @@ def parse_args(self) -> None:
...
+        argsPerHelpText = []
+        lastHelpText = list(parser._option_string_actions.values())[0]
+        for arg, helpText in parser._option_string_actions.items():
+            if(lastHelpText != helpText):
+                descriptionText = lastHelpText.help.replace('\n', '<br>')
+                print(f"| {', '.join(argsPerHelpText)} | {descriptionText} |")
+                argsPerHelpText.clear()
+
+            argsPerHelpText.append(arg)
+            lastHelpText = helpText
```
Comment (MichaelL-PA) [inline main.py:908]: sorry, the code I gave you omitted the last argument (and there's a better way to do it)

```suggestion
def print_help_formatted(self, parser):
    print("| Argument | Description |")
    print("| --- | --- |")

    allRows = []
    for arg, helpText in parser._option_string_actions.items():
        descriptionText = helpText.help.replace('\n', '<br>')
        allRows.append(f"| {', '.join(helpText.option_strings)} | {descriptionText} |")

    # remove duplicates
    for row in list(OrderedDict.fromkeys(allRows)):
        print(row)
```

you'll need to import OrderedDict too
```
from collections import OrderedDict
```

**Review by MichaelL-PA — CHANGES_REQUESTED** [review]

File: main.py [hunk 2]
```diff
@@ -775,58 +778,58 @@ def parse_args(self) -> None:
...
-            help="Set the build URI",
+            help="URI to CI build info",
             type=str,
             default=None,
             required=False,
         )
         parser.add_argument(
             "--operatingEnvironment",
-            help="Set the Operating Environment",
+            help="Operating environment for information porpuses only",
```
Comment (MichaelL-PA) [inline main.py:816]: does this return the operating environment, or set it?

**Review by MichaelL-PA — COMMENTED** [review]
I added a pull request template in .github/pull_request_template.md
```
<!---
If you've edited any of the arguments for this package:

1. Run this script with the --helpFormatted argument (ex. soos-dast --helpFormatted)
2. Copy the result and paste it in the README under '### Script Arguments'
3. Make sure your terminal didn't wrap any lines, confirm the table looks correct
-->
```

File: main.py [hunk 3]
```diff
@@ -866,13 +869,20 @@ def parse_args(self) -> None:
...
             type=bool,
             default=None,
             required=False
         )
 
+        
+
         log(f"Parsing Arguments")
+        # parse help argument
+        help: Namespace = parser.parse_args(["-h", "[--help]"])
```
Comment (MichaelL-PA) [inline main.py:882]: does this overwrite the default --help command?

Comment (SOOS-JAlvarez) [reply main.py:882]: > does this overwrite the default --help command?

Yep, thought we wanted to replace the original help behaviour but now I saw your PR

File: main.py (hunk 2 above)
Comment (SOOS-JAlvarez) [reply main.py]: changed the wording to be more clear.

**Review by MichaelL-PA — APPROVED** [review]

**Pull request merged**